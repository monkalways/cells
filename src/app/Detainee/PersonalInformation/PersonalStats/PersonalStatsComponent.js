import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    detentionUnitName: PropTypes.string.isRequired,
    dob: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    propertyBagNumber: PropTypes.string,
  }).isRequired,
};

export const PersonalStatsComponent = ({ classes, detainee }) => {
  const {
    detentionUnitName,
    dob,
    firstName,
    gender,
    lastName,
    propertyBagNumber,
  } = detainee;

  return (
    <React.Fragment>
      <Grid container spacing={0}>
        <Grid container item xs={12}>
          <Grid item xs={3}>
            <Typography variant="h6" className={classes.text}>
              Name:
            </Typography>
          </Grid>
          <Grid item xs={9}>
            <Typography variant="h6">
              {lastName}, {firstName}
            </Typography>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={3}>
            <Typography variant="h6" className={classes.text}>
              DOB:
            </Typography>
          </Grid>
          <Grid item xs={9}>
            <Typography variant="h6">{dob}</Typography>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={3}>
            <Typography variant="h6" className={classes.text}>
              Gender:
            </Typography>
          </Grid>
          <Grid item xs={9}>
            <Typography variant="h6">{gender}</Typography>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={3}>
            <Typography variant="h6" className={classes.text}>
              BIN:
            </Typography>
          </Grid>
          <Grid item xs={9}>
            <Typography variant="h6">{propertyBagNumber || <br />}</Typography>
          </Grid>
        </Grid>

        <Grid container item xs={12}>
          <Grid item xs={3}>
            <Typography variant="h6" className={classes.text}>
              Remarks:
            </Typography>
          </Grid>
          <Grid item xs={9}>
            <Typography variant="h6">{detentionUnitName}</Typography>
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

PersonalStatsComponent.propTypes = propTypes;

export default withStyles(() => ({
  text: {
    fontWeight: 400,
  },
}))(PersonalStatsComponent);
