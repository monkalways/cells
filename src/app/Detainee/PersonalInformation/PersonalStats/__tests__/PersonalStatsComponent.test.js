import React from 'react';
import { shallow } from 'enzyme';

import { PersonalStatsComponent } from '../PersonalStatsComponent';

describe('CellDetaineeCardContentComponent', () => {
  let classes;
  let detainee;

  beforeEach(() => {
    classes = {
      text: 'text',
    };
    detainee = {
      detentionUnitName: 'DMU',
      dob: '2001-01-01',
      firstName: 'Bob',
      gender: 'Other',
      lastName: 'Burger',
      propertyBagNumber: '20',
    };
  });

  const setup = () => shallow(<PersonalStatsComponent classes={classes} detainee={detainee} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
