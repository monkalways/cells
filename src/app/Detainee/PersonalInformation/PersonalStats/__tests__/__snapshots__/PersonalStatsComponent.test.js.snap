// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellDetaineeCardContentComponent should render 1`] = `
<Fragment>
  <WithStyles(Grid)
    container={true}
    spacing={0}
  >
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <WithStyles(Typography)
          className="text"
          variant="h6"
        >
          Name:
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          Burger
          , 
          Bob
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <WithStyles(Typography)
          className="text"
          variant="h6"
        >
          DOB:
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          2001-01-01
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <WithStyles(Typography)
          className="text"
          variant="h6"
        >
          Gender:
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          Other
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <WithStyles(Typography)
          className="text"
          variant="h6"
        >
          BIN:
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          20
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      item={true}
      xs={12}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <WithStyles(Typography)
          className="text"
          variant="h6"
        >
          Remarks:
        </WithStyles(Typography)>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(Typography)
          variant="h6"
        >
          DMU
        </WithStyles(Typography)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Fragment>
`;
